Source/Core/Core/IOS/IOS.h

```diff
@@ -82,6 +83,12 @@ enum ProcessId : u32
   PID_UNKNOWN = 19,
 };
 
+template <typename T>
+void WriteReturnValue(T value, u32 address)
```

Comment:
IMO, the cast should likely just be done explicitly where needed. Making all of IOS (or most of it that includes this header) dependent on Memmap means most of IOS will need to be recompiled whenever changes to the Memmap header are done.